### PR TITLE
11591 OffscreenCanvas worker thread example

### PR DESCRIPTION
**Pull request by bsmth**
This PR adds an example for using `OffscreenCanvas` in a worker thread. This is one of the main use cases of this feature. Background info here: https://wiki.whatwg.org/wiki/OffscreenCanvas#Use_Case_Description, e.g.:

* "need to be able to render WebGL from a worker, transfer the rendered image to the main thread without making any copy of it"
* "need to be able to render WebGL entirely asynchronously from a worker"
* "need to be able to layout  and render text from a worker using CanvasRenderingContext2D and display those results on the main thread."

__Notes:__
* Parent task: https://github.com/mdn/content/issues/11591

### CODE REVIEW

File: web-workers/offscreen-canvas-worker/worker.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  } else {
+    canvasB = e.data.canvas;
+    ctxWorker = canvasB.getContext("2d");
+    startCounting();
+  }
+};
+
+// Fibonacci function to add some delay to the thread
+function fibonacci(num) {
+  if (num <= 1) return 1;
```
Comment (teoli2003) [inline web-workers/offscreen-canvas-worker/worker.js:17]:
```suggestion
  if (num <= 1) {
    return 1;
  }
```

File: web-workers/offscreen-canvas-worker/worker.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  return fibonacci(num - 1) + fibonacci(num - 2);
+}
+
+// Start the counter for Canvas B
+let counter = 0;
+function startCounting() {
+  setInterval(function () {
```
Comment (teoli2003) [inline web-workers/offscreen-canvas-worker/worker.js:24]:
```suggestion
  setInterval(() => {
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
Reviewed to match the new JS guidelines 😃

File: web-workers/offscreen-canvas-worker/index.html [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        <p>
+          This demo is based on an example by
+          <a href="https://glitch.com/@PicchiKevin">Kevin Picchi</a>.
+        </p>
+      </footer>
+    </main>
+
+    <script>
+      const requestAnimationFrame = window.requestAnimationFrame;
+
```
[the remark this answers is not in the document]
Comment (bsmth) [reply web-workers/offscreen-canvas-worker/index.html:87]: well spotted

File: web-workers/offscreen-canvas-worker/index.html [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+      const canvasA = document.getElementById("canvas main");
+      const canvasB = document.getElementById("canvas worker");
+
+      const ctxA = canvasA.getContext("2d");
+      const canvasWidth = ctxA.width;
+      const canvasHeight = ctxA.height;
+
+      // Create a counter for Canvas A
+      let counter = 0;
+      setInterval(function () {
```
[the remark this answers is not in the document]
Comment (bsmth) [reply web-workers/offscreen-canvas-worker/index.html:97]: 👍🏻

File: web-workers/offscreen-canvas-worker/index.html [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      function redrawCanvasA() {
+        ctxA.clearRect(0, 0, canvasA.width, canvasA.height);
+        ctxA.font = "24px Verdana";
+        ctxA.textAlign = "center";
+        ctxA.fillText(counter, canvasA.width / 2, canvasA.height / 2);
+      }
+
+      // This function creates heavy (blocking) work on a thread
+      function fibonacci(num) {
+        if (num <= 1) return 1;
```
[the remark this answers is not in the document]
Comment (bsmth) [reply web-workers/offscreen-canvas-worker/index.html:112]: Good to know

File: web-workers/offscreen-canvas-worker/style.css [hunk 6]
```diff
@@ -0,0 +1,81 @@
+html,
+body {
+  font-family: sans-serif;
+  margin: 2rem;
+}
+
+main, footer {
```
[the remark this answers is not in the document]
Comment (bsmth) [reply web-workers/offscreen-canvas-worker/style.css]: done in https://github.com/mdn/dom-examples/pull/159/commits/144e4e15dfbef2c6082589d1545a1d97bac2b057 ✅

**Review by teoli2003 — APPROVED** [review]

**Comment by bsmth** [comment]
Thanks, merging 🙌🏻

**Pull request merged**